assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java

```diff
@@ -2224,6 +2224,24 @@ private String build_student_view_submission_confirmation_context(VelocityPortle
         context.put("value_ASSIGNMENT_INPUT_ADD_TIME_SPENT", state.getAttribute(ResourceProperties.ASSIGNMENT_INPUT_ADD_TIME_SPENT));
         context.put("isTimesheet", assignmentService.isTimeSheetEnabled((String) state.getAttribute(STATE_CONTEXT_STRING)));
 
+        try {
+            // Get site ID
+            String siteId = toolManager.getCurrentPlacement().getContext();
+            Optional<Site> site = Optional.of(siteService.getSite(siteId));
+            Site currentSite = site.get();
```

Comment:
Guys, I really wish we weren't approving and merging PRs in just a couple of hours with no one else given the time to review.

This `Optional` is unnecessary and shouldn't be here. You're just calling `.get()` on it immediately, and it's already wrapped in a try/catch for `IdUnusedException`.

Reply:
You're right Brian, but this is happening right now in other pieces of work without a single explanation, and it's my team that is suffering the consequences.

Anyway, I don't want to use this PR to complain or create conflicts about these issues, and you're right, always feel free to contact me or my team about opened or closed PRs, we'll fix or update whatever is needed, we take care of what we contribute.

We'll fix your comments on Monday.

Reply:
@druiz309 Could you work on the Optional bit please?